ontotext-yasgui-web-component/src/components/ontotext-tooltip-web-component/ontotext-tooltip-web-component.tsx

```diff
@@ -44,67 +124,122 @@ export class OntotextTooltipWebComponent {
+    const placement = (this.placement as FloatingPlacement) || 'top';
 
-    if (this.showOnClick) {
-      this.el.addEventListener('click', this.showFunction);
-    }
+    computePosition(this.el, this.popperEl, {
+      placement,
+      strategy: 'fixed',
+      middleware: [offset(8), flip(), shift({padding: 8}), arrow({element: this.arrowEl})],
+    }).then(({x, y, placement: finalPlacement, middlewareData}) => {
+      if (!this.popperEl || !this.arrowEl) return;
 
-    this.el.addEventListener('mouseleave', this.hideFunction);
-  }
+      this.popperEl.style.left = `${x}px`;
+      this.popperEl.style.top = `${y}px`;
+      this.popperEl.setAttribute('data-placement', finalPlacement);
 
-  private createShowFunction(tooltip: Instance) {
-    return () => {
-      if (this.dataTooltip) {
-        tooltip.show();
-      }
+      const ax = middlewareData.arrow?.x;
+      const ay = middlewareData.arrow?.y;
+
+      this.arrowEl.style.left = ax != null ? `${ax}px` : '';
+      this.arrowEl.style.top = ay != null ? `${ay}px` : '';
+
+      const base = finalPlacement.split('-')[0] as 'top' | 'right' | 'bottom' | 'left';
+      const staticSide = {top: 'bottom', right: 'left', bottom: 'top', left: 'right'}[base];
+      this.arrowEl.style.right = '';
+      this.arrowEl.style.bottom = '';
+      (this.arrowEl.style as Partial<CSSStyleDeclaration>)[staticSide] = '0';
+    });
+  };
+
+  private startAutoUpdate() {
+    if (this.cleanupAutoUpdate) {
+      return;
     }
+    if (this.popperEl === null) {
+      return;
+    }
+
+    // Update only on ancestor scroll/resize
+    this.cleanupAutoUpdate = autoUpdate(
+        this.el,
+        this.popperEl,
+        this.updatePosition,
+        {
+          animationFrame: false,
+          ancestorScroll: true,
+          ancestorResize: true,
+          elementResize: false
+        }
+    );
   }
 
-  private createHideFunction(tooltip: Instance) {
-    return () => tooltip.hide();
+  private stopAutoUpdate() {
+    if (this.cleanupAutoUpdate) {
+      this.cleanupAutoUpdate();
+      this.cleanupAutoUpdate = null;
+    }
   }
 
   private destroy() {
-    this.el.removeEventListener('mouseover', this.showFunction);
-    this.el.removeEventListener('click', this.showFunction);
-    this.el.removeEventListener('mouseleave', this.hideFunction);
+    if (this.onMouseOver) {
+      this.el.removeEventListener('mouseover', this.onMouseOver);
+    }
+    if (this.onMouseLeave) {
+      this.el.removeEventListener('mouseleave', this.onMouseLeave);
+    }
+    this.el.removeEventListener('click', this.show);
+
+    if (this.onDocClick) {
+      document.removeEventListener('click', this.onDocClick, true);
+      this.onDocClick = null;
+    }
+
+    if (this.popperEl) {
+      this.popperEl.removeEventListener('mouseleave', this.hide);
+      if (this.popperEl.parentNode) {
+        this.popperEl.parentNode.removeChild(this.popperEl);
+      }
+    }
+
+    this.stopAutoUpdate();
```

Comment:
The hide method contains similar logic. Please review both methods and extract the common behavior, or consider calling hide() within destroy.